saleor/plugins/invoicing/plugin.py

```diff
@@ -0,0 +1,41 @@
+from typing import Any, Optional
+from uuid import uuid4
+
+from django.core.files.base import ContentFile
+
+from ...core import JobStatus
+from ...invoice.models import Invoice
+from ...order.models import Order
+from ..base_plugin import BasePlugin
+from .utils import generate_invoice_number, generate_invoice_pdf
+
+
+class InvoicingPlugin(BasePlugin):
+    PLUGIN_ID = "mirumee.invoicing"
+    PLUGIN_NAME = "Invoicing"
+    DEFAULT_ACTIVE = True
+    PLUGIN_DESCRIPTION = "Built-in saleor plugin that handles invoice creation."
+
+    def invoice_request(
+        self,
+        order: "Order",
+        invoice: "Invoice",
+        number: Optional[str],
+        previous_value: Any,
+    ) -> Any:
+        file_content, creation_date = generate_invoice_pdf(invoice)
+        invoice.created = creation_date
+        invoice.update_invoice(number=generate_invoice_number())
+        invoice.invoice_file.save(f"{uuid4()}.pdf", ContentFile(file_content))
```

Comment:
I think it could be useful to have the order number in the file name. E.g. format `order-{order_number}-invoice-{number}` - that would be more human-readable. If we cannot guarantee the number is unique, we could then append some short UUID, like 6 characters or so.

Reply:
It might be better to stick with UUID:
- no os.path calls to check whether file exists
- some setups might use S3, would have to check if it exists on S3 in case we use S3 :fire: 
So i think it's best to do:
`order-{order_number}-invoice-{number}-{uuid4()}`
so we have both uniqueness and information in the file, then it's easier to figure what that PDF file in your downloads folder is.

Reply:
Sounds good!

Reply:
I would also stick to UUID because we don't want to risk users enumerating existing invoices by guessing their filenames. As for checking whether a file exists, we may still need to do that, UUID4 is not safe from collisions (even if they are unlikely, see the birthday paradox).